plugins/woocommerce/client/blocks/assets/js/blocks/product-gallery/frontend.ts

```diff
@@ -67,6 +67,73 @@ const scrollImageIntoView = ( imageId: number ) => {
 	}
 };
 
+/**
+ * Scrolls the thumbnail into view.
+ *
+ * @param {number} imageId - The ID of the thumbnail to scroll into view.
+ */
+const scrollThumbnailIntoView = ( imageId: number ) => {
+	if ( ! imageId ) {
+		return;
+	}
+
+	// Get the current element that triggered the action
+	const element = getElement()?.ref as HTMLElement;
+
+	if ( ! element ) {
+		return;
+	}
+
+	// Find the closest gallery container
+	const galleryContainer = element.closest(
+		'.wp-block-woocommerce-product-gallery'
+	);
+
+	if ( ! galleryContainer ) {
+		return;
+	}
+
+	const thumbnailElement = galleryContainer.querySelector(
+		`.wc-block-product-gallery-thumbnails__thumbnail img[data-image-id="${ imageId }"]`
+	);
+
+	if ( ! thumbnailElement ) {
+		return;
+	}
+
+	// Find the thumbnail scrollable container
+	const scrollContainer = thumbnailElement.closest(
+		'.wc-block-product-gallery-thumbnails__scrollable'
+	);
+
+	if ( ! scrollContainer ) {
+		return;
+	}
+
+	const thumbnail = thumbnailElement.closest(
+		'.wc-block-product-gallery-thumbnails__thumbnail'
+	);
+
+	if ( ! thumbnail ) {
+		return;
+	}
+
+	// Calculate the scroll position to center the thumbnail
+	const containerRect = scrollContainer.getBoundingClientRect();
+	const thumbnailRect = thumbnail.getBoundingClientRect();
+
+	const scrollTop =
+		scrollContainer.scrollTop +
+		( thumbnailRect.top - containerRect.top ) -
+		( containerRect.height - thumbnailRect.height ) / 2;
+
+	// Use scrollTo to avoid scrolling the entire page
+	scrollContainer.scrollTo( {
+		top: scrollTop,
+		behavior: 'smooth',
+	} );
```

Comment:
This logic assumes Thumbnails are vertical but they can be horizontal as well:
- it can be changed manually by switching Gallery Area from Row to Stack
- or such layout is forced on mobile (thumbnails are pushed below large image).

https://github.com/user-attachments/assets/cdf580c9-9e63-4350-8649-86211dc2afcf

Wouldn't `scrollIntoView` solve it for both cases without calculating the offset? Unless it's needed to center the thumbnails?

Reply:
I see what you meant that we have to use `scrollTo` to avoid page scroll. I extended your approach to cover horizontal thumbnails layout: https://github.com/woocommerce/woocommerce/pull/57195/commits/4f61fc7488e1882182c71b085dd1f4501d4b129a